articles/implementing-network-address-translation/index.md

```diff
@@ -0,0 +1,205 @@
+
+### Requirement.
+This tutorial uses the CISCO packet tracer [Download packet tracer](https://www.netacad.com/portal/resources/packet-tracer).
+*NOTE To better understand this tutorial, the reader is expected to have a  foundational knowledge of IPV4 and the CISCO command-line interface (CLI)*
+
+### Internet Protocol
+According to [Wikipeida](https://en.wikipedia.org/wiki/IP_address), an Internet Protocol address (IP address) is a numerical label assigned to each device connected to a computer network that uses the Internet Protocol for communication.
+IP address currently have two version in used: Internet Protocol version 4 (IPv4) and Internet Protocol Version 6 (IPv6)
+IPv4 defines an IP address as a 32-bit number while IPv6 defines IP address as a 128-bit number
+### Public and private ip address
+All IPv4 addresses can be divided into two groups: public (global) and private (local) addresses.
+Public addresses are routable addresses that are used on the internet.
+These addresses allow users to access resources on a computer network located anywhere in the world.
+Private addresses are not routable  and no traffic can be sent to them or by them over the internet.
+These address are within the range : 10.0.0.0 to 10.255.255.255.255, 172.16.0.0 to 172.255.255.255 and 192.168.0.0 t0 192.168.255.255
+
+### Private addressing.
+The major limitation of Internet Protocol Version 4 (IPV4) was its address exhaustion issue.
+As a short-term solution, various protocols such as private addressing and Network Address Translation (NAT) were introduced.
+These two standards work closely together, allowing organizations to assign private addresses to their internal network while translating them to public addresses to allow them connect to the internet.
+Some devices in an organization's network may not need to connect to the internet; when building such a network, an administrator is expected to use private IP addresses as defined in [RFC 1918](https://tools.ietf.org/html/rfc1918).
+The documentation defines a set of network addresses assigned to an organization's internal network so that devices can communicate locally.
+If there is a need for such devices to connect to the internet, their private addresses must be translated to public addresses using Network Address Translation (NAT).
+
+### Network Address Translation
+Because private addresses are not routable addresses, for a device configured with a private address to access the internet or a remote network, the address must be translated into a public routable address.
+This translation takes place on a NAT-enabled router which typically operates on the border of a stub network.
+![Network address translation](/engineering-education/implementing-network-address-translation/nat.jpg)
+In the figure above, PCA with an IP address of 172.31.1.2 want to reach the webserver, but because PCA's address is not routable, because of this it cannot access the server directly.
```

Comment:
```suggestion
In the figure above, PCA with an IP address of 172.31.1.2 wants to reach the webserver, but because PCA's address is not routable, because of this it cannot access the server directly.
```